tests/foreman/ui/test_rhc.py

```diff
@@ -0,0 +1,163 @@
+"""Test class for RH Cloud connector - rhc
+
+:Requirement: Remoteexecution
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: RHCloud-CloudConnector
+
+:Assignee: jpathan
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+from fauxfactory import gen_string
+from wait_for import wait_for
+
+from robottelo import constants
+from robottelo.config import settings
+from robottelo.constants import DEFAULT_SUBSCRIPTION_NAME
+from robottelo.helpers import file_downloader
+from robottelo.logging import logger
+
+
+@pytest.fixture(scope='module')
+def fixture_enable_rhc_repos(module_target_sat):
+    """Enable repos required for configuring RHC."""
+    # subscribe rhc satellite to cdn.
+    module_target_sat.register_to_cdn()
+    if module_target_sat.os_version.major == 8:
+        module_target_sat.enable_repo(constants.REPOS['rhel8_bos']['id'])
+        module_target_sat.enable_repo(constants.REPOS['rhel8_aps']['id'])
+    else:
+        module_target_sat.enable_repo(constants.REPOS['rhscl7']['id'])
+        module_target_sat.enable_repo(constants.REPOS['rhel7']['id'])
+
+
+@pytest.fixture(scope='module')
+def module_rhc_org(module_target_sat):
+    """Module level fixture for creating organization"""
+    org = module_target_sat.api.Organization(
+        name=settings.rh_cloud.organization or gen_string('alpha')
+    ).create()
+    # adding remote_execution_connect_by_ip=Yes at org level
+    module_target_sat.api.Parameter(
+        name='remote_execution_connect_by_ip',
+        value='Yes',
+        organization=org.id,
+    ).create()
+    return org
+
+
+@pytest.fixture()
+def fixture_setup_rhc_satellite(request, module_target_sat, module_rhc_org):
+    """Create Organization and activation key after successful test execution"""
+    yield
+    if request.node.rep_call.passed:
```

Comment:
I am just curious, why do you need to perform this whole setup after the test is finished?

Reply:
It's to address stage 2 work mentioned in the SAT-9327 jira card.
Basically, it'll be used for running iqe tests.